src/json2typescript/json-convert.ts

```diff
@@ -869,73 +869,68 @@ export class JsonConvert {
 
         }
 
-        // Check if attempt and expected was n-d
-        if (expectedJsonType instanceof Array && value instanceof Array) {
-
-            let array: any[] = [];
-
-            // No data given, so return empty value
-            if (value.length === 0) {
-                return array;
-            }
-
-            // We obviously don't care about the type, so return the value as is
-            if (expectedJsonType.length === 0) {
-                return value;
+        // Check if expected was n-d
+        if (expectedJsonType instanceof Array) {
+            if (value === null) {
+                if (this.valueCheckingMode !== ValueCheckingMode.DISALLOW_NULL) return null;
+                else throw new Error("\tReason: Given value is null.");
             }
 
-            // Loop through the data. Both type and value are at least of length 1
-            let autofillType: boolean = expectedJsonType.length < value.length;
-            for (let i = 0; i < value.length; i++) {
-
-                if (autofillType && i >= expectedJsonType.length) expectedJsonType[i] = expectedJsonType[i - 1];
-
-                array[i] = this.verifyProperty(expectedJsonType[i], value[i], serialize);
+            // Check that value is not primitive
+            if (value instanceof Object) {
+                let array: any[] = [];
 
-            }
+                // No data given, so return empty value
+                if (value.length === 0) {
+                    return array;
+                }
 
-            return array;
+                // We obviously don't care about the type, so return the value as is
+                if (expectedJsonType.length === 0) {
+                    return value;
+                }
+                // Copy the expectedJsonType array so we don't change the class-level mapping based on the value of this property
+                const jsonType: any[] = expectedJsonType.slice(0);
```

Comment:
So basically this is the only part that is changed, apart from line 899/903/096 when actually using the value.

If I see correctly, your changes in this method are just a restructuring of the if-else logic?

Reply:
Yes - the key fix is to use slice() to copy the expectedJsonType instead of using it directly, since it gets modified.  

I restructured the if-else branches since they seemed to be doing a lot of the same stuff across multiple branches.